backend/prompt_studio/prompt_studio_core_v2/serializers.py

```
@@ -44,48 +44,70 @@ class Meta:
                 str(instance.tool_id),
             )
-        prompt_instance: ToolStudioPrompt = ToolStudioPrompt.objects.filter(
+
+        # Fetch prompt instances
+        prompt_instances = ToolStudioPrompt.objects.filter(
             tool_id=data.get(TSKeys.TOOL_ID)
         ).order_by("sequence_number")
-        data[TSKeys.PROMPTS] = []
+
+        if not prompt_instances.exists():
+            data[TSKeys.PROMPTS] = []
+            return data
+
+        # Process prompt instances
         output: list[Any] = []
-        # Appending prompt instances of the tool for FE Processing
-        if prompt_instance.count() != 0:
-            for prompt in prompt_instance:
-                profile_manager_id = prompt.prompt_id
-                if instance.single_pass_extraction_mode:
-                    # use projects default profile
-                    profile_manager_id = profile_manager.profile_id
-                prompt_serializer = ToolStudioPromptSerializer(prompt)
+        for prompt in prompt_instances:
+            prompt_serializer = ToolStudioPromptSerializer(prompt)
+            serialized_data = prompt_serializer.data
+
+            # Determine coverage
+            coverage: list[Any] = []
+            profile_manager_id = prompt.profile_manager
+            if default_profile and instance.single_pass_extraction_mode:
+                profile_manager_id = default_profile.profile_id
+
+            if profile_manager_id:
                 coverage = OutputManagerUtils.get_coverage(
                     data.get(TSKeys.TOOL_ID),
                     profile_manager_id,
                     prompt.prompt_id,
                     instance.single_pass_extraction_mode,
                 )
-                serialized_data = prompt_serializer.data
-                serialized_data["coverage"] = coverage
-                output.append(serialized_data)
-            data[TSKeys.PROMPTS] = output
+            else:
+                logger.info(
```

@jagadeeswaran-zipstack under what scenario can this happen? Why would a profile ID go missing?
Is it when we delete all profiles in a project? Ideally this should not be possible at all - 
1. we should throw an error if there's only 1 profile and the user tries to delete it
1. Or we should throw an error early if there's no profile and a user tries to run a prompt

---

@chandrasekharan-zipstack Default profile and only 1 profile can't be deleted. This is for old projects where users might have deleted all the project. we will change this logic in future.

---

Sure :+1: